packages/next/build/webpack-config.ts

```diff
@@ -160,6 +160,7 @@ export function getDefineEnv({
     'process.env.__NEXT_TEST_MODE': JSON.stringify(
       process.env.__NEXT_TEST_MODE
     ),
+    'process.env.__NEXT_PROD_BUILD': JSON.stringify(!dev),
```

Comment:
Where is this needed that `NODE_ENV === 'production'` can't be used?

Reply:
Since `dev` can be directly specified via https://nextjs.org/docs/advanced-features/custom-server, it's technically possible that `dev` is `false` but `NODE_ENV` is missing (although I think that will be a misuse).

Reply:
We should be showing a warning for that case currently here https://github.com/vercel/next.js/blob/980095d59e0f108019415641a724910cd41a7630/packages/next/server/next.ts#L183 and I think that could cause other issues like the wrong package version being loaded and such

Reply:
Yeah will do that change instead 👍